common-libs/common/src/services/DidDocumentService/ConcreteDidDocument.ts

```diff
@@ -0,0 +1,7 @@
+import { DidResolver } from '../../shared/DidResolver'
+
+export interface ConcreteDidDocument<T extends { id: string } = { id: string }> {
```

Comment:
Not sure why do we need this generic parameter here. It would probably be enough to declare the return value of `buildDidDocument` as `Record<string, unknown> & { id: string }`.

Also I'm not sure about the naming.
This is not a did document; this is a did document factory? wrapper? something else? Did document is something that `buildDidDocument` returns, but not the object with that method.
And this is an interface, as far from "concrete" as it could be.